### DISCUSSION TITLE
How can I initially load routes/upstream/services from a file with docker

### DISCUSSION THREAD

**Discussion by abbluiz** [apache/apisix#5208] [Q&A] [answered]
I wanted to use a infrastructure as code approach to configuring some routes in the gateway. Is there a YAML/JSON file that the gateway can read during docker build (or with docker volumes) so that I can commit some default routes to git, for example?

**Comment by tzssangglass** [comment, marked as answer]
try stand-alone mode:https://github.com/apache/apisix/blob/master/docs/en/latest/stand-alone.md

**Reply by abbluiz** [reply]
Excellent. Last question: do I have to disable etcd?

**Reply by tzssangglass** [reply]
stand-alone mode does not require etcd
